### PR TITLE
A Deeper Look at Git: Update for Style Guide

### PR DESCRIPTION
**Pull request by mgrigoriev8109**
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

**1. Because:**
The Deeper Look at Git lesson did not adhere to the new Style Guide

**2. This PR:**
Updated the lesson stylistically to adhere to the Style Guide

**3. Additional Information:**
Two important changes were also made:
1.  the merge conflicts section was moved into being an Assignment, due to its use of outside links.
2.  the think-like-a-git resource was included in the Assignment section, due to the fact that it fits well in the lesson when compared to its current location in the [Using Git in the Real World lesson](https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/using-git-in-the-real-world)

### CODE REVIEW

File: ruby_programming/git/lesson_a_deeper_look_at_git.md [hunk 1]
```diff
@@ -140,26 +141,19 @@ So now that we've learned about the various dangerous of `git push --force`, you
...
-  - For `git push --force` only use it when appropriate, use it with caution, and preferably default to using `git push --force-with-lease`
+1.  If working on a team project make sure rewriting history is safe to do, and others know you're doing it.</span>
+2.  Ideally stick to using these commands only on branches that you're working with by yourself.
+3.  Using the `-f` flag to force something should scare you, and you better have a really good reason for using it.
+4.  Don't push after every single commit, changing published history should be avoided when possible
+5.  Regarding the specific commands we've covered:
+  1.  For `git amend` never amend commits that have been pushed to remote repositories
+  2.  For `git rebase` never rebase a repository that others may work off of 
+  3.  For `git reset` never reset commits thath have been pushed to remote repositories
+  4.  For `git push --force` only use it when appropriate, use it with caution, and preferably default to using `git push --force-with-lease`
```
[the remark this answers is not in the document]
Comment (mgrigoriev8109) [reply ruby_programming/git/lesson_a_deeper_look_at_git.md]: Fixed it! Doesn't display properly in https://www.theodinproject.com/lessons/preview

So confirming in TOP's #contribution-suggestions-bugs-discussions channel that it's simply an issue with the preview tool.

**Review by wise-king-sullyman — APPROVED** [review]
🔥 this is ready to merge, but I'm going to hold it and merge at the same time that my Using Git in the Real World PR gets merged so since an assignment is being moved from there to here.

**Comment by mgrigoriev8109** [comment]
@wise-king-sullyman the merge conflict has been resolved.

**Comment by wise-king-sullyman** [comment]
🔥 🔥 🔥 merging!

**Pull request merged**